qemu/tests/hugepage_reset.py

```diff
@@ -0,0 +1,74 @@
+from avocado.utils import process
+
+from virttest import env_process
+from virttest import test_setup
+from virttest import utils_test
+from virttest import utils_misc
+from virttest.staging import utils_memory
+
+
+def run(test, params, env):
+    """
+    Re-assign nr-hugepages
+
+    1. Set up hugepage with 1G page size and hugepages=N
+    2. Boot up guest using /mnt/hugetlbfs/ as backend in QEMU CML
+    3. Change the nr_hugepages after the guest boot up (M < N and M > N)
+    4. Run the stress test **inside guest**
+    5. change the nr_hugepages after the stress test (M < N and M > N)
+
+    :param test: QEMU test object.
+    :param params: Dictionary with test parameters.
+    :param env: Dictionary with the test environment.
+    """
+
+    def set_hugepage():
+        """Set nr_hugepages"""
+        process.run('echo 1 > /proc/sys/vm/compact_memory')
```

Comment:
This cmd can add to pre_command in cfg, with drop caches cmd.
pre_command = "echo 3 > /proc/sys/vm/drop_caches && echo 1 > /proc/sys/vm/compact_memory"

Reply:
@huangyum Do we need to clean it every time before boot up guest, cause sometimes the first round boot up(M<N) succeed but the second time(M>N) it failed.

Reply:
We only need boot up guest once. N, M is the number of hugepages we set before and after guest boot up.

Reply:
@huangyum I don't really understand, now the step 3 and step 5 all say (M>N and M<N) so it looks like we have 2 sub scenario, we need to boot with a smaller M before and after guest boot, and then a larger M before and after guest boot. Now you mean we only need to boot it once, like before boot set a smaller M(6) and after boot a larger M(10)?

Reply:
M is not the guest memory. 
Let's set N=8, guest memory=4G, then change hugepage numbers by set M. For M>N, set M=10, for M<N, set M=6.

Reply:
@huangyum Yes I understand that it's not guest memory but the hugepage number to be set. According to the test case description, it seems to be:
1. Set the hugepage number to 8
2. Boot guest -> Set hugepage number to 6 AND 10
3. Stress guest -> Set hugepage number to 6 AND 10
How do I handle this `AND`? My understanding is:
Boot guest-> Set it to 10 -> Stress guest -> Set it to 10 -> Destroy guest -> Boot guest -> Set it to 6 -> Stress guest -> Set it to 6.
If only boot it once, do you mean: Boot guest -> Set it to 10 -> Set to 6 ->Stress guest -> Set to 10 -> Set to 6?

Reply:
Yes, it's the latter one.
Boot guest -> Set it to 10 -> Set to 6 ->Stress guest -> Set to 10 -> Set to 6.

Reply:
@huangyum I see, thanks for the clarification, will update soon.